### ISSUE TITLE
Wrong file owner in image pulled from Docker Hub

### ISSUE THREAD

**Issue by tothtamas28** [runtimeverification/kontrol#42]
Before pushing the image on CI `kontrol --help` worked fine: https://github.com/runtimeverification/kontrol/actions/runs/6273447948/job/17036929997

After pulling the image:

```
$ docker run --rm runtimeverificationinc/kontrol:ubuntu-jammy-0.1.3 kontrol --help
Unable to find image 'runtimeverificationinc/kontrol:ubuntu-jammy-0.1.3' locally
ubuntu-jammy-0.1.3: Pulling from runtimeverificationinc/kontrol
445a6a12be2b: Pull complete
5283d12e99e9: Pull complete
9b6d1a784f13: Pull complete
eabe052c5de0: Pull complete
b95de52dfae6: Pull complete
3425cedef1dd: Pull complete
803108da4639: Pull complete
8d286995a433: Pull complete
e5701859d8ff: Pull complete
4f4fb700ef54: Pull complete
2b91b930eb52: Pull complete
3e7ae33f2b82: Pull complete
be278fa6099f: Pull complete
Digest: sha256:56a95ab151dde546a8c9089ae7fd8b9b76638f66b025b2a916a8f5687164cd45
Status: Downloaded newer image for runtimeverificationinc/kontrol:ubuntu-jammy-0.1.3

$ docker images
REPOSITORY                       TAG                  IMAGE ID       CREATED        SIZE
runtimeverificationinc/kontrol   ubuntu-jammy-0.1.3   d62a63dc5d84   27 hours ago   2.93GB

$ docker run --rm runtimeverificationinc/kontrol:ubuntu-jammy-0.1.3 kontrol --help
/usr/bin/python3: can't open file '/home/user/.local/bin/kontrol': [Errno 13] Permission denied
```

This is because all files and folders in `/home/user` are owned by `root`:

```
$ docker run --rm --user root runtimeverificationinc/kontrol:ubuntu-jammy-0.1.3 ls -la
total 32
drwxr-x--- 1 root root 4096 Sep 21 09:31 .
drwxr-xr-x 1 root root 4096 Sep 21 09:20 ..
-rw-r--r-- 1 root root  220 Jan  6  2022 .bash_logout
-rw-r--r-- 1 root root 3771 Jan  6  2022 .bashrc
drwxr-xr-x 4 root root 4096 Sep 21 09:25 .cache
drwxr-xr-x 4 root root 4096 Sep 21 09:20 .foundry
drwxr-xr-x 4 root root 4096 Sep 21 09:25 .local
-rw-r--r-- 1 root root  807 Jan  6  2022 .profile
```

When building the image locally, file ownership is set properly to `user`. (File ownership is probably set properly also on CI, but this needs checking.)

**Comment by tothtamas28** [comment]
Might be related: https://github.com/runtimeverification/evm-semantics/pull/2068#issuecomment-1720954112

**Comment by tothtamas28** [comment]
@F-WRunTime, have you seen anything similar before?

**Comment by yale-vinson** [comment]
@F-WRunTime Can I assign this to you and move it to In Progress?

**Comment by F-WRunTime** [comment]
> @F-WRunTime Can I assign this to you and move it to In Progress?

Yes

**Comment by F-WRunTime** [comment]
Wild... After pushing to dockerhub the permissions are no longer correct. 
I've checked direct on a runner the image that got pushed 
![image](https://github.com/runtimeverification/kontrol/assets/105403280/afd56602-2fc7-4595-87be-f42393cea65c)

**Comment by F-WRunTime** [comment]
Run on two different machines. Was built on -9 and pulled down by -24 
![image](https://github.com/runtimeverification/kontrol/assets/105403280/e6806cf1-4ff3-4b08-b1e5-f5f049b41059)

New test. Built on -9 pushed to internal registry. Same issue?

**Comment by tothtamas28** [comment]
Btw, image `v0.1.9` works for me:

```
% docker run --rm runtimeverificationinc/kontrol:ubuntu-jammy-0.1.9 kontrol --help
Unable to find image 'runtimeverificationinc/kontrol:ubuntu-jammy-0.1.9' locally
...
Digest: sha256:af8e9a4d0f7b3a051486ae1857c3d0e038c0064c82745e661c99a92f88b1bab9
Status: Downloaded newer image for runtimeverificationinc/kontrol:ubuntu-jammy-0.1.9
usage: kontrol [-h]
               {version,compile,solc-to-k,build,prove,show,to-dot,list,view-kcfg,remove-node,simplify-node,step-node,merge-nodes,section-edge,get-model}
...
```

**Comment by F-WRunTime** [comment]
Between 10 <> 9 
![image](https://github.com/runtimeverification/kontrol/assets/105403280/28e89ef5-8e4c-42aa-bd57-a840083974ad)

Their base layers are different. 
![image](https://github.com/runtimeverification/kontrol/assets/105403280/3a824b01-b1fd-4a74-aeda-70168d115b46)


Will revert any changes to the dockerfile to 0.1.9 build and determine if there's a change there.

**Comment by F-WRunTime** [comment]
- [ ] https://github.com/runtimeverification/kontrol/actions/runs/6339829440/job/17219843401

**Comment by F-WRunTime** [comment]
This PR published a viable image for use 
https://github.com/runtimeverification/kontrol/pull/59 -- This can be closed.

**Comment by tothtamas28** [comment]
Fixed in #59.

**Comment by tothtamas28** [comment]
Issue surfaced in `runtimeverificationinc/kontrol:ubuntu-jammy-0.1.22` once again.

**Comment by F-WRunTime** [comment]
Alright after many hours testing / debugging it seems I have found a few problematic runners causing the issue. Specifically WHY they are producing images such that the overlay2 folders on pull down from remote repository is incorrect isn't clear. 

I rebuilt the known broken runner I could find. And redeployed them. I standardized the runners installation of buildx on all runners, as it was not clear it was installed everywhere properly. Waiting on next build of Kontrol to see if there is improvement.
